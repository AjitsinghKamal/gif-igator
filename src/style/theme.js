import palette from './palette';
export const FONT_BASE = 16;
const COLORS = Object.keys(palette).reduce((theme, colorKey) => {
	const meta = colorKey.split('_');
	const themeType = meta[1] || 'main';
	if (themeType in theme) {
		theme[themeType][meta[0]] = palette[colorKey];
	} else {
		theme[themeType] = { [meta[0]]: palette[colorKey] };
	}
	return theme;
}, {});

export default Object.assign({ fontBase: FONT_BASE }, COLORS);
