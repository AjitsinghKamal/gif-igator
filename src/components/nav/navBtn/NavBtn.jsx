import React from 'react';
import { EmText } from 'style/Helpers';
import styled from 'styled-components';
import PropTypes from 'prop-types';
const Btn = styled.button`
	display: block;
	border: 0;
	height: 160px;
	width: 100%;
	border-radius: 20px;
	background: pink;
	margin: 15px;
`;
const NavBtn = ({ title }) => {
	return (
		<Btn>
			<EmText sz={25} wt={800}>
				{title}
			</EmText>
		</Btn>
	);
};
NavBtn.propTypes = {
	theme: PropTypes.object,
	title: PropTypes.string,
};
export default NavBtn;
