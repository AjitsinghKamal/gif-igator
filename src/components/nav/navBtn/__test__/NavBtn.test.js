import React from 'react';
import NavBtn from '../NavBtn';
import { shallow } from 'enzyme';
import theme from 'style/theme';
describe('Navbtn component', () => {
	it('renders correctly in dark theme', () => {
		const darktheme = Object.assign({}, theme.base, theme.dark);
		const wrapper = shallow(<NavBtn theme={darktheme} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('renders correctly in white theme', () => {
		const lighttheme = Object.assign({}, theme.base, theme.light);
		const wrapper = shallow(<NavBtn theme={lighttheme} />);
		expect(wrapper).toMatchSnapshot();
	});
});
