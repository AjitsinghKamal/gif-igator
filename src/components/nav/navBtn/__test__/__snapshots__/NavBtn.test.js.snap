// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbtn component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Navbtn component renders correctly in dark theme 1`] = `ShallowWrapper {}`;

exports[`Navbtn component renders correctly in white theme 1`] = `ShallowWrapper {}`;

exports[`Searchbox component renders correctly 1`] = `ShallowWrapper {}`;
